setup.py

```diff
@@ -25,38 +29,59 @@ def package_files(directory):
     "../mlflow/temporary_db_migrations_for_pre_1_users/alembic.ini",
 ]
 
+"""
+Minimal requirements for the skinny MLflow client which provides a limited
+subset of functionality such as: RESTful client functionality for Tracking and
+Model Registry, as well as support for Project execution against local backends
+and Databricks.
+"""
+SKINNY_REQUIREMENTS = [
+    "click>=7.0",
+    "cloudpickle",
+    "databricks-cli>=0.8.7",
+    "entrypoints",
+    "gitpython>=2.1.0",
+    "numpy",
+    "pandas",
+    "python-dateutil",
+    "pyyaml",
+    "protobuf>=3.6.0",
+    "requests>=2.17.3",
+    "six>=1.10.0",
+    "simplejson",
+]
+
+_is_mlflow_skinny = bool(os.environ.get(_MLFLOW_SKINNY_ENV_VAR))
```

Comment:
Nit: Can we move this down to line 78-ish (right above `setup()`). The var naming + comments look awesome - thanks so much for the investment there!